### PR TITLE
@filipesilva fix(@angular-devkit/build-angular): validate fileReplacements, allow string styles/scripts

### PR DESCRIPTION
**Pull request by filipesilva**
Fix angular/angular-cli#5053
Fix angular/angular-cli#10267

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/browser/schema.json [hunk 1]
```diff
@@ -67,10 +67,17 @@
           "src": {
```
Comment (hansl) [inline packages/angular_devkit/build_angular/src/browser/schema.json]: Could you `oneOf` this instead? With this current schema we could have `{ replaceWith: "foo" }` which is clearly invalid.

File: packages/schematics/angular/migrations/update-6/index.ts [hunk 2]
```diff
@@ -279,7 +279,7 @@ function extractProjectsConfig(config: CliConfig, tree: Tree): JsonObject {
             ...(isProduction && serviceWorker ? { serviceWorker: true } : {}),
             fileReplacements: [
               {
-                src: `${app.root}/${source}`,
+                source: `${app.root}/${source}`,
```
Comment (hansl) [inline packages/schematics/angular/migrations/update-6/index.ts]: The form I'd like to see is `{ replace: "some/file/path", with: "some/other/file" }` which to mean feels more natural. What do you think?

File: packages/angular_devkit/build_angular/src/browser/schema.json (hunk 1 above)
Comment (filipesilva) [reply packages/angular_devkit/build_angular/src/browser/schema.json]: Yeah that is better.

File: packages/schematics/angular/migrations/update-6/index.ts (hunk 2 above)
Comment (filipesilva) [reply packages/schematics/angular/migrations/update-6/index.ts]: I'm ok with that.

File: packages/angular_devkit/build_angular/src/browser/schema.d.ts [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+  /**
+   * Defines the optimization level of the build.
+   */
+  optimization: boolean;
+
+  /**
+   * Replace files with other files in the build.
+   */
+  fileReplacements: FileReplacements[];
```
Comment (larssn) [inline packages/angular_devkit/build_angular/src/browser/schema.d.ts]: Don't think this type - 'FileReplacements' - made into the source. I've looked around for it, but haven't found it.

Comment (filipesilva) [reply packages/angular_devkit/build_angular/src/browser/schema.d.ts]: @larssn it did make it in, it's something we test for in https://github.com/angular/devkit/blob/master/packages/angular_devkit/build_angular/test/browser/replacements_spec_large.ts

**Pull request merged**